web3/middleware/filter.py

```diff
@@ -235,14 +244,19 @@ def __init__(
 
     @property
     def from_block(self) -> BlockNumber:
-        return self._from_block
+        if is_string(self._from_block) and is_hex(self._from_block):
```

Comment:
we should probably be normalizing the `self._from_block` value before assigning it which should make this un-necessary.